roles/ocp_add_users/tasks/add-users.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Set all the users (new and current)
+  vars:
+    current_users: "{{ (_oau_secret.resources[0].data.htpasswd | default('Cg==') | b64decode).split() }}"
+    new_users: "{{ (_oau_new_users_encoded_password.content | b64decode).split() }}"
+  ansible.builtin.set_fact:
+    oau_all_users: "{{ _oau_all_users | default({}) | combine({item.split(':')[0]: item.split(':')[1]}) }}"
+  loop: "{{ current_users + new_users }}"
+  loop_control:
+    label: "{{ item.split(':')[0] }}"
+  no_log: "{{ oau_secure_log | bool }}"
+
+- name: Create/Update htpasswd secret
+  vars:
+    _all_users_text: |
+      {% for user, passwd in oau_all_users.items() %}
+      {{ user + ":" + passwd }}
+      {% endfor %}
+  community.kubernetes.k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: htpass-secret
+        namespace: openshift-config
+      type: Opaque
+      data:
+        htpasswd: "{{ _all_users_text | b64encode }}"
+  no_log: "{{ oau_secure_log | bool }}"
+
+- name: Setup htpasswd auth IdP backend in OCP
```

Comment:
this should only be done when `oau_idp_type == HTPasswd`

Reply:
In fact I think this whole block (maybe the entire role?) works only with HTPasswd, not sure why have it as a variable

Reply:
Yeah... I thought it may be possible to include another type later... But looks like the other IdPs are remote...

So while I do define a var, it is only one type that is supported *and* validated, I made sure I *did not* document it in the README, so there is no expectation to tweak that value.

If it is too confusing I can remove it no problem.

Reply:
I did remove the var, no more confusion :+1: